ui/src/app/pages/admin/admin-workspace-search.tsx

```diff
@@ -0,0 +1,62 @@
+import {Component} from '@angular/core';
+import {Button} from 'app/components/buttons';
+import {FlexRow} from 'app/components/flex';
+import {TextInput} from 'app/components/inputs';
+import colors from 'app/styles/colors';
+import {ReactWrapperBase, UrlParamsProps, withUrlParams} from 'app/utils';
+import {navigate} from 'app/utils/navigation';
+import * as React from 'react';
+
+interface State {
+  workspaceNamespace: string;
+}
+
+class AdminWorkspaceSearchImpl extends React.Component<UrlParamsProps, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      workspaceNamespace: '',
+    };
+  }
+
+  navigateToWorkspace() {
```

Comment:
This I object to more than the above comment. Navigating away from the search bar means that you have to go back in order to search a second thing. I would strongly prefer to keep the search bar and do something with state that forces a fetch of the workspace information when the router is sent to `admin/workspaces/{namespace}`.

Reply:
You can easily return to the search form via the browser "back" command.

I do agree that this isn't an ideal UX, but it's an intermediate step on the way to a longer-term goal, where the "search" page actually shows search results, and then you click into those results to open up a new tab with each workspace's detail view. I think that setup would address your concerns, but we can chat offline if you're worried it won't.

Reply:
This proooobably would. Especially if we want to do different kinds of search, like workspaces-by-problem-user or something.